ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh

```diff
@@ -6,98 +6,133 @@ set -o pipefail
 
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
 
-# Make sure yq is installed
-if [ ! -f /tmp/yq-v4 ]; then
-  # TODO move to image
-  curl -L "https://github.com/mikefarah/yq/releases/download/v4.25.3/yq_linux_$(uname -m | sed s/aarch64/arm64/ | sed s/x86_64/amd64/)" -o /tmp/yq-v4 && chmod +x /tmp/yq-v4
-  PATH=${PATH}:/tmp
-fi
+function get_ready_nodes_count() {
+  oc get nodes \
+    -o jsonpath='{range .items[*]}{.metadata.name}{","}{.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}' | \
+    grep -c -E ",True$"
+}
+
+# wait_for_nodes_readiness loops until the number of ready nodes objects is equal to the desired one
+function wait_for_nodes_readiness()
+{
+  local expected_nodes=${1}
+  local max_retries=${2:-10}
+  local period=${3:-5}
+  for i in $(seq 1 "${max_retries}") max; do
+    if [ "${i}" == "max" ]; then
+      echo "[ERROR] Timeout reached. ${expected_nodes} ready nodes expected, found ${ready_nodes}... Failing."
+      return 1
+    fi
+    sleep "${period}m"
+    ready_nodes=$(get_ready_nodes_count)
+    if [ x"${ready_nodes}" == x"${expected_nodes}" ]; then
+        echo "[INFO] Found ${ready_nodes}/${expected_nodes} ready nodes, continuing..."
+        return 0
+    fi
+    echo "[INFO] - ${expected_nodes} ready nodes expected, found ${ready_nodes}..." \
+      "Waiting ${period}min before retrying (timeout in $(( (max_retries - i) * (period) ))min)..."
+    (( max_retries -- ))
```

Comment:
if it does timeout , do we collect must-gather ?

Reply:
good point, let me check. We might need to delete some file in the SHARED_DIR.

Reply:
regardless of that, I don't need the line you are commenting on :D

Reply:
removed the line and add this other to allow the `gather` chain to do its job.

Reply:
https://github.com/openshift/release/pull/35101/files#diff-9e75860c660730e2cb77472d634a02aa6da64e7bac073d6cbfb7daf4133188a1R138

Reply:
looks good!

Reply:
As discussed offline, we don't need it. must-gather is taken in any case. The exit code in the install-status.txt file will feed the junits to report the installation status. As far as we keep the scale-out of the additional workers as a day2 operation, we're good to go without it. Resolving the thread